vms/platformvm/state/state_test.go

```diff
@@ -109,620 +110,384 @@ func TestStateSyncGenesis(t *testing.T) {
-					addPermValTx.ID(),
-					utxVal,
-					time.Unix(valStartTime, 0),
-					validatorReward,
-				)
-				r.NoError(err)
+	tests := map[string]struct {
+		initialStakers []*Staker
+		initialTxs     []*txs.Tx
 
-				utxDel := createPermissionlessDelegatorTx(subnetID, delegatorData)
-				addPermDelTx := &txs.Tx{Unsigned: utxDel}
-				r.NoError(addPermDelTx.Initialize(txs.Codec))
+		// Staker to insert or remove
+		staker      *Staker
+		addStakerTx *txs.Tx // If tx is nil, the staker is being removed
 
-				del, err := NewCurrentStaker(
-					addPermDelTx.ID(),
-					utxDel,
-					time.Unix(delStartTime, 0),
-					delegatorReward,
-				)
-				r.NoError(err)
+		// Check that the staker is duly stored/removed in P-chain state
+		expectedCurrentValidator  *Staker
+		expectedPendingValidator  *Staker
+		expectedCurrentDelegators []*Staker
+		expectedPendingDelegators []*Staker
 
-				r.NoError(s.PutCurrentValidator(val))
-				s.AddTx(addPermValTx, status.Committed) // this is currently needed to reload the staker
-				r.NoError(s.Commit())
+		// Check that the validator entry has been set correctly in the
+		// in-memory validator set.
+		expectedValidatorSetOutput *validators.GetValidatorOutput
 
-				s.PutCurrentDelegator(del)
-				s.AddTx(addPermDelTx, status.Committed) // this is currently needed to reload the staker
-				r.NoError(s.Commit())
-				return del
-			},
-			checkStakerInState: func(r *require.Assertions, s *state, staker *Staker) {
-				delIt, err := s.GetCurrentDelegatorIterator(staker.SubnetID, staker.NodeID)
-				r.NoError(err)
-				r.True(delIt.Next())
-				retrievedDelegator := delIt.Value()
-				r.False(delIt.Next())
-				delIt.Release()
-				r.Equal(staker, retrievedDelegator)
+		// Check whether weight/bls keys diffs are duly stored
+		expectedWeightDiff    *ValidatorWeightDiff
+		expectedPublicKeyDiff maybe.Maybe[*bls.PublicKey]
+	}{
+		"add current primary network validator": {
+			staker:                   primaryNetworkCurrentValidatorStaker,
+			addStakerTx:              addPrimaryNetworkValidator,
+			expectedCurrentValidator: primaryNetworkCurrentValidatorStaker,
+			expectedValidatorSetOutput: &validators.GetValidatorOutput{
+				NodeID:    primaryNetworkCurrentValidatorStaker.NodeID,
+				PublicKey: primaryNetworkCurrentValidatorStaker.PublicKey,
+				Weight:    primaryNetworkCurrentValidatorStaker.Weight,
+			},
+			expectedWeightDiff: &ValidatorWeightDiff{
+				Decrease: false,
+				Amount:   primaryNetworkCurrentValidatorStaker.Weight,
 			},
-			checkValidatorsSet: func(r *require.Assertions, s *state, staker *Staker) {
-				val, err := s.GetCurrentValidator(staker.SubnetID, staker.NodeID)
-				r.NoError(err)
-
-				valsMap := s.validators.GetMap(staker.SubnetID)
-				r.Contains(valsMap, staker.NodeID)
-				valOut := valsMap[staker.NodeID]
-				r.Equal(valOut.NodeID, staker.NodeID)
-				r.Equal(valOut.Weight, val.Weight+staker.Weight)
+			expectedPublicKeyDiff: maybe.Some[*bls.PublicKey](nil),
```

Comment:
why would adding a validator not update the `publicKeyDiff`?

Reply:
This is specifying a key diff (providing a `maybe.Some` means there is a key diff present). Adding a validator results in a key diff where the prior value was `nil`.